language/move-prover/src/prelude.bpl

```diff
@@ -327,6 +327,10 @@ function {:inline} $IsNormalizedValueArray(a: $ValueArray, len: int): bool {
     (forall i: int :: i < 0 || i >= len ==> v#$ValueArray(a)[i] == $DefaultValue())
 }
 
+function {{backend.func_inline}} $ContainValueArray(a: $ValueArray, v: $Value): bool {
+    (var len := l#$ValueArray(a);
+     (exists i: int :: i < len && $IsEqual(v#$ValueArray(a)[i], v)))
```

Comment:
This should be `i >= 0 && i < len`.

Reply:
I corrected the condition. That makes me need to assume that the length of ValueArray is 0 (or at least non-negative). So, I added the assumption that the EventStore is initially empty in `$InitVerification()` in Prelude.